compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -423,6 +425,7 @@ impl<'tcx> ConstToPat<'tcx> {
                         tcx.sess.emit_err(err);
 
                         // FIXME: introduce PatKind::Error to silence follow up diagnostics due to unreachable patterns.
+                        // We errored, so the pattern we generate is irrelevant.
                         PatKind::Wild
```

Comment:
Should this one set `saw_const_match_error` too?

Reply:
Hm, it doesn't check for `saw_const_match_error` either. So I guess yes? Not sure. This PR was not meant to have any behavior changes in it, just comments and  clarifications.

Reply:
Ah fair; I was just thinking that should also trigger a `delay_span_bug`

Reply:
The `emit_err` is directly above here so theres'a definitely a hard error. In the other case where I added a `delay_span_bug`, the `emit_err` is conditional on something, so it becomes a non-local property, and I felt it was worth simplifying the reasoning, so that the failure mode for if `saw_const_match_error` gets set incorrectly is an ICE rather than a weird miscompilation.